File: App/Core/Environment.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Runtime.CompilerServices;

namespace GLSharp.Core {
    [Imported]
    [IgnoreNamespace]
    public class Environment : IEnvironment{
        public float[] CreateFloat32Array(ulong size) {
            throw new global::System.Exception("Not implemented");
        }

        public object Init(object obj, object[] args) {
            throw new global::System.Exception("Not implemented");
        }
    }
}

File: App/Scripts/Environment.js
var Environment = function () {
}

Environment.prototype = {
    createFloat32Array: function (param) {
        return new Float32Array(param);
    },

    init: function(obj, args) {
        for(var i = 0; i < args.length - 1; i += 2)
            obj[args[i]] = args[i + 1];

        return obj;
    }
}